### ISSUE TITLE
CA1052 (Static holder types) should not special-case empty interfaces

### ISSUE THREAD

**Issue by sharwell** [dotnet/roslyn-analyzers#1163]
#### Analyzer package

Example: Microsoft.CodeQuality.Analyzers

#### Analyzer

CA1052 (Static holder types)

#### Repro steps

1. Declare a non-static type which contains a static method and implements an empty interface

```csharp
public interface IMarker
{
}

public class Marker : IMarker
{
    public static void Foo() { }
}
```

#### Expected behavior

CA1052 should not be reported.

#### Actual behavior

CA1052 is reported.

➡️ If marker interfaces are a bad practice, that should be implemented as a separate rule. However, in the absence of such a rule it is not reasonable to assume that marker interfaces are inherently bad, or that the use of a marker interface should be predicated on meeting one of the following seemingly arbitrary conditions:

1. The implementer has an unrelated non-static member
2. The implementer has no static members

**Comment by 333fred** [comment]
This behavior was explicitly introduced in #280. @srivatsn or @genlu, any comments on this?

**Comment by srivatsn** [comment]
What does FxCop do? It sounds like #280 basically added the check to match FxCop behavior? We tried to keep breaking changes to a minimum - however in this case the break would be in the opposite direction - FxCop would complains and we wouldn't in which case it might be reasonable.

**Comment by 333fred** [comment]
Yeah, FxCop reports here. @dotnet/roslyn-analysis, any thoughts? I agree with this change, and I'm fine with adding it, since we aren't adding new warning locations.

**Comment by sharwell** [comment]
💭 I felt that this change has the potential to avoid a small number of false positives in the wild without negatively impacting the primary purpose of this analyzer. Given that one of the primary target audiences is users who *didn't* use FXCop previously, the attention to false positives will be a boost to overall adoption rates and retention.

**Comment by jinujoseph** [comment]
Sounds reasonable to me.
